storage/cache/gen.go

```diff
@@ -21,11 +21,10 @@ import (
 	"context"
 
 	"github.com/google/trillian/storage/storagepb"
-	"github.com/google/trillian/storage/tree"
 )
 
 // NodeStorage provides an interface for storing and retrieving subtrees.
 type NodeStorage interface {
-	GetSubtree(n tree.NodeID) (*storagepb.SubtreeProto, error)
+	GetSubtree(prefix []byte) (*storagepb.SubtreeProto, error)
```

Comment:
Unfortunately golang has no immutable byte string type. There could be value in using string instead. OTOH the original NodeID type was not immutable and I suspect it's not worth doing.

Reply:
Within the scope of this PR, the safety remains the same as it was. I might improve it in follow-ups.